source/base/index_set.cc

```diff
@@ -514,10 +514,8 @@ IndexSet::make_trilinos_map (const MPI_Comm &communicator,
   // we risk to call different Epetra_Map on different processors and the code
   // hangs.
 #ifdef DEAL_II_WITH_MPI
-  bool all_contiguous = is_contiguous();
-  MPI_Allreduce(MPI_IN_PLACE, &all_contiguous, 1, MPI_LOGICAL, MPI_LAND,
-                communicator);
-  if ((all_contiguous == true) && (!overlapping))
+  bool all_contiguous = (Utilities::MPI::min (is_contiguous() ? 1 : 0, communicator) == 1);
+  if ((all_contiguous) && (!overlapping))
 #else
   if ((is_contiguous() == true) && (!overlapping))
 #endif
```

Comment:
I'm not a huge fan of having `#ifdef` interrupt regular control flow. But I think you don't even need it here -- `Utilities::MPI::min` should also work without MPI and just return the original argument. So just go with the `#if` case here.

Also make the `all_contiguous` variable `const`, if you can.